great_expectations/core/usage_statistics/anonymizers/base.py

```diff
@@ -0,0 +1,692 @@
+import copy
+import logging
+from abc import ABC, abstractmethod, abstractstaticmethod
+from hashlib import md5
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+from great_expectations.core.usage_statistics.util import (
+    aggregate_all_core_expectation_types,
+)
+from great_expectations.util import deep_filter_properties_iterable, load_class
+
+from great_expectations.core.usage_statistics.anonymizers.types.base import (  # isort:skip
+    GETTING_STARTED_DATASOURCE_NAME,
+    GETTING_STARTED_EXPECTATION_SUITE_NAME,
+    GETTING_STARTED_CHECKPOINT_NAME,
+    BATCH_REQUEST_REQUIRED_TOP_LEVEL_KEYS,
+    BATCH_REQUEST_OPTIONAL_TOP_LEVEL_KEYS,
+    DATA_CONNECTOR_QUERY_KEYS,
+    RUNTIME_PARAMETERS_KEYS,
+    BATCH_SPEC_PASSTHROUGH_KEYS,
+    BATCH_REQUEST_FLATTENED_KEYS,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseAnonymizer(ABC):
```

Comment:
Take all shared behavior from `Anonymizer` and place here - this is subclassed throughout. All child classes must fulfill the two abstract methods.